scheduler/src/workflow-execution.ts

```diff
@@ -89,21 +90,26 @@ async function executeStorage (args: { pipelineConfig: PipelineConfig, data: obj
   return dataLocation
 }
 
-async function executeNotification (args: { pipelineConfig: PipelineConfig, dataLocation: string, data?: object, error?: object }): Promise<void> {
+async function executeNotification (args: { pipelineConfig: PipelineConfig; dataLocation: string; data?: object; error?: object }): Promise<void> {
+  await AmqpClient.init()
   const pipelineConfig = args.pipelineConfig
   const data = args.data
   const error = args.error
 
-  const notificationTrigger: NotificationClient.NotificationTriggerEvent = {
+  const notificationTrigger: NotificationTriggerEvent = {
     dataLocation: args.dataLocation,
     pipelineId: pipelineConfig.id,
     pipelineName: pipelineConfig.metadata.displayName,
     data: data,
     error: error
   }
 
-  await NotificationClient.executeNotification(notificationTrigger)
-  console.log(`Successfully delivered notification request to transformation-service for ${pipelineConfig.id}`)
+  const success = AmqpClient.publish(notificationTrigger)
```

Comment:
await I think

Reply:
The amqplib we use only offers a sync publish method. I was surprised by that too but decided to keep the wrapper asynchronous as well. Do you think I should wrap it asynchronously or try to find a way to have the publication acked?

Reply:
Getting an ACK is good, even better if it would work async... but just make it work for the time being, there is still time for those optimizations afterwards